### PR TITLE
Refactor EInkDisplay

### PR DESCRIPTION
**Pull request by todd-herbert**
### What

In consultation with @markbirss,
* Merges *some* variant-specific code
* ~~Shifts macros to respective `variant.h` files~~
  Shifts macros to respective `platformio.ini` files
  Previously tested displays documented at top of `EInkDisplay2.cpp`
* Change some global variables in `EInkDisplay2.cpp` to members of the `EInkDisplay` class
* Purges all "dynamic refresh" code

### Why

General tidiness, but also in preparation to re-implement "dynamic refresh" as a derived class of EInkDisplay (easier to maintain, easier to scrap).
More tidying could probably be done. The variant-specific code in `connect()` has been completely avoid, for fear of breaking something important in some horrible way.

### Who this might affect

Heltec Wireless Paper V1.0 reverts to using the stock "full-refresh" E-Ink system of updates.

Minor refactoring affects a **large amount of boards**. In theory, no changes should be noticed. I'm *reasonably* sure that everything went into the right `variant.h` files, **but it would be good to get a second pair of eyes on this.**

### CODE REVIEW

**Comment by todd-herbert** [comment]
Something spectacular going wrong with one `#include`. I'll get back to you.

File: variants/rak4631/variant.h [hunk 1]
```diff
@@ -133,6 +133,24 @@ static const uint8_t SCK = PIN_SPI_SCK;
 #define PIN_EINK_SCLK (0 + 3)
 #define PIN_EINK_MOSI (0 + 30) // also called SDI
 
+// E-Ink model selection
+
+#define EINK_DISPLAY_MODEL GxEPD2_213_BN // 2.13 inch b/w 250x122
+#define EINK_WIDTH 250
+#define EINK_HEIGHT 122
+
+// #define EINK_DISPLAY_MODEL GxEPD2_420_M01 // 4.2 inch 300x400
```
Comment (todd-herbert) [inline variants/rak4631/variant.h:142]: In cases like this, would you like me to bring the commented-out boards into the platformio.ini file also, as grouped sets of build flags?

```ini
build_flags = 
  ; 2.13 inch b/w 250x122 
  -DEINK_DISPLAY_MODEL=GxEPD2_213_BN
  -DEPD_WIDTH=250
  -DEPD_HEIGHT=122

  ; 4.2 inch 300x400
  ;-DEINK_DISPLAY_MODEL=GxEPD2_420_M01
  ;-DEPD_WIDTH=300
  ;-DEPD_HEIGHT=400
```
(etc)

**Comment by markbirss** [comment]
@todd-herbert

We should use consistent names eg
EINK_DISPLAY_MODEL
EINK_WIDTH
EINK_HEIGHT

File: variants/rak4631/variant.h (hunk 1 above)
Comment (markbirss) [reply variants/rak4631/variant.h:142]: No we can consolidate those just into a single list to keep as known tested or used displays variants

**Comment by todd-herbert** [comment]
Getting a build error for `meshtastic-diy-v1`, but than seems related to [4ffb906](https://github.com/meshtastic/firmware/commit/4ffb906fe8f45cb40e628ad0edbe146670d725ed)

File: variants/m5stack_coreink/variant.h
Comment (markbirss) [inline variants/m5stack_coreink/variant.h:1]: #define EINK_DISPLAY_MODEL GxEPD2_154_M09 // 1.54 inch
#define EINK_WIDTH 200
#define EINK_HEIGHT 200

should be commented out/removed here, already defined in platformio.ini ?

File: variants/monteops_hw1/platformio.ini
Comment (markbirss) [inline variants/monteops_hw1/platformio.ini:1]: no need to remove RAK4630 here ?

File: variants/rak10701/variant.h
Comment (markbirss) [inline variants/rak10701/variant.h:1]: #define EINK_DISPLAY_MODEL GxEPD2_213_BN // 2.13 inch b/w 250x122
#define EINK_WIDTH 250
#define EINK_HEIGHT 122

should be commented out/removed here, already defined in platformio.ini ?

File: variants/rak4631/variant.h
Comment (markbirss) [inline variants/rak4631/variant.h:1]: #define EINK_DISPLAY_MODEL GxEPD2_213_BN // 2.13 inch b/w 250x122
#define EINK_WIDTH 250
#define EINK_HEIGHT 122

should be commented out/removed here, already defined in platformio.ini ?

File: variants/rak4631_epaper/variant.h
Comment (markbirss) [inline variants/rak4631_epaper/variant.h:1]: #define EINK_DISPLAY_MODEL GxEPD2_213_BN // 2.13 inch b/w 250x122
#define EINK_WIDTH 250
#define EINK_HEIGHT 122

should be commented out/removed here, already defined in platformio.ini ?

File: variants/rak4631_epaper_onrxtx/variant.h
Comment (markbirss) [inline variants/rak4631_epaper_onrxtx/variant.h:1]: #define EINK_DISPLAY_MODEL GxEPD2_213_BN // 2.13 inch b/w 250x122
#define EINK_WIDTH 250
#define EINK_HEIGHT 122

should be commented out/removed here, already defined in platformio.ini ?

File: variants/t-echo/variant.h
Comment (markbirss) [inline variants/t-echo/variant.h:1]: #define EINK_DISPLAY_MODEL GxEPD2_154_D67
#define EINK_WIDTH 200
#define EINK_HEIGHT 200

should be commented out/removed here, already defined in platformio.ini ?

File: variants/Dongle_nRF52840-pca10059-v1/variant.h
Comment (markbirss) [inline variants/Dongle_nRF52840-pca10059-v1/variant.h:1]: #define EINK_DISPLAY_MODEL GxEPD2_420_M01 // 4.2 inch 300x400
#define EINK_WIDTH 300
#define EINK_HEIGHT 400

should be commented out/removed here, already defined in platformio.ini ?

File: variants/m5stack_coreink/platformio.ini
Comment (markbirss) [inline variants/m5stack_coreink/platformio.ini:1]: #define EINK_DISPLAY_MODEL GxEPD2_213_BN
#define EINK_WIDTH 250
#define EINK_HEIGHT 122

should be commented out/removed here, already defined in platformio.ini ?

File: variants/heltec_wireless_paper_v1/variant.h
Comment (markbirss) [inline variants/heltec_wireless_paper_v1/variant.h:1]: #define EINK_DISPLAY_MODEL GxEPD2_213_BN
#define EINK_WIDTH 250
#define EINK_HEIGHT 122

should be commented out/removed here, already defined in platformio.ini ?

**Review by markbirss — COMMENTED** [review]
changes look good, just update the variant.h files as we only need them defined in platformio.ini not both

File: variants/monteops_hw1/platformio.ini
Comment (todd-herbert) [reply variants/monteops_hw1/platformio.ini:1]: Yeah.. sorry. I don't know how it got deleted, some slip of the finger when using find and replace?
I swear, I did catch this and fix it though, before I made a huge mess of the commit..

**Review by markbirss — APPROVED** [review]

**Comment by todd-herbert** [comment]
> @todd-herbert t-echo trunk issue ?

Huh.. weird. I'm out of the house right now but I'll look into it in a little bit.

**Review by thebentern — APPROVED** [review]

**Pull request merged**